pkg/render/monitor/monitor.go

```diff
@@ -222,7 +226,24 @@ func (mc *monitorComponent) Objects() ([]client.Object, []client.Object) {
 		toCreate = append(toCreate, mc.prometheusOperatorPodSecurityPolicy())
 	}
 
+	if mc.cfg.Monitor.ExternalPrometheus != nil {
+		toCreate = append(toCreate, mc.externalConfigMap())
+		if mc.cfg.ExternalPrometheus && mc.cfg.Monitor.ExternalPrometheus.ServiceMonitor != nil {
+			externalServiceMonitor, needsRBAC := mc.externalServiceMonitor()
+			toCreate = append(toCreate, externalServiceMonitor)
+			if needsRBAC {
+				toCreate = append(toCreate, mc.externalPrometheusRole(), mc.externalPrometheusRoleBinding(), mc.externalServiceAccount(), mc.externalPrometheusTokenSecret())
```

Comment:
Do we need to `toDelete` these if ExternalPrometheus is nil?

Reply:
This is not possible, because if ExternalPrometheus is nil, we don't know what namespace to delete it from.

Reply:
The way to do this would be with a finalizer, so that we have an opportunity to clean up before the object disappears.

Reply:
We do this in the user controller already and it's fairly straightforward: https://github.com/tigera/operator/blob/d54d499ca96085a38878a21996da2d5eff1f3cf5/pkg/controller/logstorage/users/users_controller.go#L400-L406